lib/rubygems/resolver.rb

```diff
@@ -111,6 +111,7 @@ def initialize(needed, set = nil)
     @skip_gems           = {}
     @soft_missing        = false
     @stats               = Gem::Resolver::Stats.new
+    @find_possible       = {}
```

Comment:
Is this memoization for performance necessary for the fix? Can you split it to a separate PR and provide a benchmark proving that it makes a difference?

Reply:
ping @MSP-Greg

Reply:
As I stated in the comment for the commit:
Using `Gem::RequestSet.new(<rails (>= 0.0.0)>).resolve`,
`find_possible` was called 447 times, but 398 calls were previously calculated.

Maybe instead of a benchmark, try setting a breakpoint in #find_possible for debugging...

Reply:
@MSP-Greg I can take care of checking how much of a difference the memoization makes but, can you split it out from this PR?

Reply:
> can you split it out from this PR

The 2nd commit is built on top of it.  So, I'd prefer not to.

Given that the code involved will be hitting Rubygems.org, I doubt the net time change is even noticeable.  I was debugging in the method, and 447 vs 49 is an order of magnitude drop...

Reply:
Ok, I'll do it.